prowler/providers/aws/services/iam/lib/policy.py

```diff
@@ -68,3 +68,52 @@ def is_policy_public(policy: dict) -> bool:
                     ) and "Condition" not in statement:
                         return True
     return False
+
+
+def check_full_service_access(service: str, policy: dict) -> bool:
+    """
+    check_full_service_access checks if the policy allows full access to a service.
+    Args:
+        service (str): The service to check.
+        policy (dict): The policy to check.
+    Returns:
+        bool: True if the policy allows full access to the service, False otherwise.
+    """
+
+    full_access = False
+
+    if policy:
+        policy_statements = policy.get("Statement", [])
+
+        if not isinstance(policy_statements, list):
+            policy_statements = [policy["Statement"]]
```

Comment:
Add tests for the missing lines in this file please.

Reply:
Added here: https://github.com/prowler-cloud/prowler/pull/4255/commits/68f2cc51b814503e31a8592d57f6370e9298fbd5